src/Symfony/Component/Validator/Tests/Mapping/ClassMetadataTest.php

```diff
@@ -308,6 +309,19 @@ public function testGetPropertyMetadataReturnsEmptyArrayWithoutConfiguredMetadat
     {
         $this->assertCount(0, $this->metadata->getPropertyMetadata('foo'), '->getPropertyMetadata() returns an empty collection if no metadata is configured for the given property');
     }
+
+    public function testClassMetadataMembersAreUniqueByPropertyAndByClassAfterMergingConstraints()
+    {
+        $metadata = new ClassMetadata(self::INTERFACE_A_CLASS);
+        $metadata->addGetterConstraint('data', new ConstraintA());
+
+        $otherMetadata = new ClassMetadata(self::INTERFACE_A_CLASS);
+        $otherMetadata->addGetterConstraint('data', new ConstraintB());
+
+        $metadata->mergeConstraints($otherMetadata);
+
+        $this->assertCount(1, $metadata->getPropertyMetadata('data'));
```

Comment:
maybe you're right.. we cannot foresee the possible things people are doing. I dont like the idea of a deep comparison nightmare; see #15950 which we need to consider as well.

I think we actually need to prevent the wrong call to `mergeConstraint` based on  the above inheritance.